lib/ipam/interface.go

```diff
@@ -41,8 +41,8 @@ type Interface interface {
 	ReleaseIPs(ctx context.Context, ips []cnet.IP) ([]cnet.IP, error)
 
 	// GetAssignmentAttributes returns the attributes stored with the given IP address
-	// upon assignment.
-	GetAssignmentAttributes(ctx context.Context, addr cnet.IP) (map[string]string, error)
+	// upon assignment, as well as the handle used for assignment (if any).
+	GetAssignmentAttributes(ctx context.Context, addr cnet.IP) (map[string]string, *string, error)
```

Comment:
Why don't we return an `AllocationAttribute` struct?

I think it would be unfortunate to use different terminology on this interface compared with the model.  E.g. "assignment attribute" = `AllocationAttribute.AttrSecondary` and "the handle used for assignment" = `AllocationAttribute.AttrPrimay`

Reply:
Yeah.. the `AllocationAttribute` is an internal representation right now. 

I agree it would be nice to align this API with our models, but I think that is outside the scope of this change. This interface is already defined in different terminology than our internal models - it uses attributes and handles. For example, that's what callers of this interface pass in when allocating an address: https://github.com/projectcalico/libcalico-go/blob/c9b2cded4f4c1fae08477795ea451429a499cdcf/lib/ipam/ipam_types.go#L28-L33